### ISSUE TITLE
Date Literal Rewrite

### ISSUE THREAD

**Issue by rossknudsen** [rossknudsen/Vba.Language#1]
# Date Literal Rewrite discussion

When you enter a date literal into the VBE, it will convert it to a standard format (if necessary).  This format is similar to the following options:
- `#MM/dd/YYYY#`
- `#hh:mm:ss AM#`
- `#MM/dd/YYYY hh:mm:ss AM#`

The specification allows for a wider range of options for expressing date and time literals but these alternatives are always converted to the above formats.  

The question is, as always, what formats should be permitted by this grammar?  Should we allow only the VBE rewritten options, or should we follow the specification?

**Comment by rubberduck203** [comment]
@rossknudsen what does the spec have to say? Are they all acceptable, but the VBA team decided to transform them for simplicity perhaps? Could be wise to follow their lead.

**Comment by rossknudsen** [comment]
The spec allows for the following things:
- Date separators include '/', ',' and '-'.
- Time separators include '.' and ':'
- Months can be expressed like 'february', 'feb' and '2'.
- And the date and time component values can be expressed in different orders and some are optional.

All of these get rewritten to the format in the original post.

Why the VBE devs did that, I have no idea.  It could be simplicity as you've suggested, in which case making my grammar simpler is desirable.  Obviously code that has been authored through the VBE will be in the simpler format.

Maybe, I should just comment out the grammar sufficiently to accommodate the simpler format and then if in the future it becomes necessary to support the alternatives, it will be easy to implement?

**Comment by rubberduck203** [comment]
Maybe. I think (again) it comes down to the goals of the project. Do you want to be able to parse code that came out of the VBE, or all valid VBA? Personally, I think dealing with stuff that comes out of the VBE is sufficient.

**Comment by rossknudsen** [comment]
You're right, that does make all the difference to the decision.  I've already decided that, for version 1 at least, to support the grammar that the VBE accepts.  So its my reference implementation.  So based on your last comment, I think we are on the same page.

Also, I have updated the project Readme to include this goal, so there is context for these kind of discussions.  I'm expecting to post a few over the next wee while.
